### ISSUE TITLE
Reproducing RoBERTa Pre-Training from Examples

### ISSUE THREAD

**Issue by pvcastro** [pytorch/fairseq#1085]
I'm trying to reproduce the RoBERTa pre-training example from 
https://github.com/pytorch/fairseq/blob/master/examples/roberta/README.pretraining.md
And the training ended right after epoch 180 finished, with the following logs:

```
| epoch 180:    543 / 546 loss=1.753, ppl=3.37, wps=43626, ups=0, wpb=214422.600, bsz=512.000, num_updates=98132, lr=0.000116817, gnorm=0.523, clip=0.000, oom=0.000, loss_scale=16.000, wall=490568, train_wall=450575
| epoch 180:    544 / 546 loss=1.753, ppl=3.37, wps=43627, ups=0, wpb=214426.000, bsz=512.000, num_updates=98133, lr=0.000116813, gnorm=0.523, clip=0.000, oom=0.000, loss_scale=16.000, wall=490573, train_wall=450580
| epoch 180:    545 / 546 loss=1.753, ppl=3.37, wps=43626, ups=0, wpb=214208.939, bsz=511.472, num_updates=98134, lr=0.000116809, gnorm=0.524, clip=0.000, oom=0.000, loss_scale=16.000, wall=490575, train_wall=450582

| epoch 180 | loss 1.753 | ppl 3.37 | wps 43626 | ups 0 | wpb 214208.939 | bsz 511.472 | num_updates 98134 | lr 0.000116809 | gnorm 0.524 | clip 0.000 | oom 0.000 | loss_scale 16.000 | wall 490575 | train_wall 450582
```
I was expecting that the training would finish only after updating for 125.000 times, as it's set for TOTAL_UPDATES. Since I got only 98.134 updates, and learning rate was still 0.000116809, why did the training stopped?

I also didn't get the log from fairseq_cli/train.py, in line 97:

`print('| done training in {:.1f} seconds'.format(train_meter.sum))`

Here's the command I used:

`fairseq-train --fp16 $DATA_DIR     --task masked_lm --criterion masked_lm     --arch roberta_base --sample-break-mode complete --tokens-per-sample $TOKENS_PER_SAMPLE     --optimizer am --adam-betas '(0.9, 0.98)' --adam-eps 1e-6 --clip-norm 0.0     --lr-scheduler polynomial_decay --lr $PEAK_LR --warmup-updates $WARMUP_UPDATES --total-num-update $TOTAL_UPDATES     --dropout 0.1 --attention-dropout 0.1 --weight-decay 0.01     --max-sentences $MAX_SENTENCES --update-freq $UPDATE_FREQ     --max-update $TOTAL_UPDATES --log-format simple --log-interval 1 --ddp-backend=no_c10d 2>/dev/null`

The logged parameters were:

`Namespace(activation_dropout=0.0, activation_fn='gelu', adam_betas='(0.9, 0.98)', adam_eps=1e-06, arch='roberta_base', attention_dropout=0.1, best_checkpoint_metric='loss', bpe=None, bucket_cap_mb=25, clip_norm=0.0, cpu=False, criterion='masked_lm', curriculum=0, data='data-bin/wikitext-103', dataset_impl=None, ddp_backend='no_c10d', device_id=0, disable_validation=False, distributed_backend='nccl', distributed_init_method='tcp://localhost:16750', distributed_no_spawn=False, distributed_port=-1, distributed_rank=0, distributed_world_size=2, dropout=0.1, encoder_attention_heads=12, encoder_embed_dim=768, encoder_ffn_embed_dim=3072, encoder_layers=12, end_learning_rate=0.0, find_unused_parameters=False, fix_batches_to_gpus=False, force_anneal=None, fp16=True, fp16_init_scale=128, fp16_scale_tolerance=0.0, fp16_scale_window=None, freq_weighted_replacement=False, keep_interval_updates=-1, keep_last_epochs=-1, leave_unmasked_prob=0.1, log_format='simple', log_interval=1, lr=[0.0005], lr_scheduler='polynomial_decay', mask_prob=0.15, mask_whole_words=False, max_epoch=0, max_sentences=16, max_sentences_valid=16, max_tokens=None, max_tokens_valid=None, max_update=125000, maximize_best_checkpoint_metric=False, memory_efficient_fp16=False, min_loss_scale=0.0001, min_lr=-1, no_epoch_checkpoints=False, no_last_checkpoints=False, no_progress_bar=False, no_save=False, no_save_optimizer_state=False, num_workers=1, optimizer='adam', optimizer_overrides='{}', pooler_activation_fn='tanh', pooler_dropout=0.0, power=1.0, random_token_prob=0.1, required_batch_size_multiple=8, reset_dataloader=False, reset_lr_scheduler=False, reset_meters=False, reset_optimizer=False, restore_file='checkpoint_last.pt', sample_break_mode='complete', save_dir='checkpoints', save_interval=1, save_interval_updates=0, seed=1, sentence_avg=False, skip_invalid_size_inputs_valid_test=False, task='masked_lm', tbmf_wrapper=False, tensorboard_logdir='', threshold_loss_scale=None, tokenizer=None, tokens_per_sample=512, total_num_update=125000, train_subset='train', update_freq=[16], use_bmuf=False, user_dir=None, valid_subset='valid', validate_interval=1, warmup_updates=10000, weight_decay=0.01)`

Thanks!

**Comment by huihuifan** [comment]
Looks strange, if the experiment finishes due to total number of updates, it should print that it finished training in N seconds. Are you sure the job did not finish due to other issues? Can you check your error logs?

**Comment by pvcastro** [comment]
Hi @huihuifan ! 
I didn't keep a log for this, just got printed to console. And I had to attach a '2>/dev/null' to the end of the command because there were too many warnings from pytorch. Do you think this '2>/dev/null' may have sent the error messages away? 
Any other place I could look for logs?
Thanks!

**Comment by myleott** [comment]
Sorry, without the log we can't debug this further, but please reopen if this happens again!

**Comment by nghuyong** [comment]
@myleott I am also trying to reproduce roberta on wiki-103, how can I check I trained successfully? Is there any metric for this model, such as dev loss ? Thanks
